src/nuketesting/_cli/runner.py

```diff
@@ -0,0 +1,165 @@
+
+    def _clean_executable_args(self) -> None:
+        """Check and clean the executable args."""
+        if "-t" in self._executable_args:
+            index = self._executable_args.index("-t")
+            self._executable_args.pop(index)
+
+    @staticmethod
+    def _check_nuke_executable(executable: Path) -> None:
+        """Check that the nuke path is a valid path on the current system.
+
+        Args:
+            nuke_path: path to the nuke installation directory.
+
+        Raises:
+            RunnerException if path is not executable.
+        """
+        if not executable.exists():
+            msg = f"Provided nuke path '{executable}' does not exist."
+            raise RunnerException(msg)
+
+    def _find_nuke_python_package(self) -> Path | None:
+        """Find the Nuke Python package based on the executable directory.
+
+        Raises:
+            RunnerException: when run on MacOS
+
+        Returns:
+            path to site packages of Nuke.
+        """
+        if platform.system().lower() == "darwin":
+            msg = "On MacOS the tests can only run in terminal mode."
+            raise RunnerException(msg)
+        python_version = sys.version_info
+        packages_path = (
+            self._nuke_executable.parent
+            / "lib"
+            / f"python{python_version.major}.{python_version.minor}"
+            / "site-packages"
+        )
+        if not packages_path.is_dir():
+            msg = (
+                "The current python version does not match the Nuke version. "
+                f"This is Python: {python_version.major}.{python_version.minor}. "
+                "Please run in terminal mode instead."
+            )
+            raise RunnerException(msg)
+        return packages_path
+
+    def execute_tests(self, test_path: str | Path) -> int:
+        """Run the testrunner with provided arguments.
+
+        Args:
+            test_path: filepath to the tests. Can be relative to the current working directory.
+                       Individual tests can be executed with the file.py::TestClass::test_function
+                       syntax. For more details consult the pytest documentation.
+        """
+        return self._execute_in_nuke(test_path) if self._run_in_terminal_mode else self._execute_native(test_path)
+
+    def _get_packages_directory(self) -> str:
+        """Get the PATH to the packages locations necessary for running tests."""
+        packages_directory = Path(pytest.__file__).parent.parent
+        testrunner_directory = Path(nuketesting.__file__).parent.parent
+        return f"{packages_directory!s}:{testrunner_directory!s}"
+
+    def _execute_in_nuke(self, test_path: str | Path) -> int:
+        """Execute the tests using the Nuke interpreter.
+
+        Args:
+            test_path: path to tests
+
+        Returns:
+            int: exitcode of tests
+        """
+        packages_directory = self._get_packages_directory()
+        arguments = [
```

Comment:
Much nicer 👍  I really appreciate your work here. Also the subprocess call is much cleaner.